internal/metrics/remote_write.go

```diff
@@ -458,6 +467,45 @@ func (r *RemoteWrite) removeServerCaFiles() {
 	}
 }
 
+// skipEmptyRanges
+// Used for skipping ranges that are outside the DB blocks and head
+// Given a point in time (less than maxTSDB) find the time of the closest block (or head)
+// Return timeVal if timeVal is within a block (or head)
+// Otherwise, return the MinTime of the closest block (or head)
+func (r *RemoteWrite) skipEmptyRanges(timeVal time.Time) time.Time {
+	result := time.Time{}
+
+	blocks := r.tsdb.BlocksMinMax()
+
+	for i := 0; i < len(blocks)-1; i += 2 {
+		minTime := blocks[i]
+		maxTime := blocks[i+1]
+
+		if timeVal.Sub(maxTime) > 0 { // timeVal ahead of this block
+			continue
+		}
+
+		if timeVal.Sub(minTime) < 0 { // timeVal precedes this block
+			result = minTime
+		} else { // timeVal contained by this block
+			result = timeVal
+		}
+
+		break
+	}
+
+	if result.IsZero() { // no block is ahead or contains timeVal. Check head
+		minTime := r.tsdb.HeadMinTime()
```

Comment:
Just a nit, I would remove the if/else  in there, and I' would write something simpler: 


```
 if result.IsZero() { // no block is ahead or contains timeVal. Check head
      minTime := r.tsdb.HeadMinTime()
      if timeVal.Sub(minTime) >= 0 { 
        return timeVal
      }
}
return result
```

Reply:
you mean
`resukt = r.tsdb.HeadMinTime()
`
and just '>' for the if.
OK

Reply:
done